### PR TITLE
Issues/552

### LINKED ISSUE #552: Group based host scan with timeout
## Specify type:
- Enhancement

## Description:
- When performing the host scan break the inventory into groups=10 systems.
- Run the playbook multiple times for each of the created groups copying the fact output into a sequential result files (e.g. *variables_path* + groupNumber)
- Run host scan with a 30 minute timeout. Hitting the timeout essentially means we are going to skip that group of hosts (e.g. no output dictionary variable file for a failed group). Log systems affected by timeout.
- After all groups have been run collect JSON results from each of the existing output files and process the data for groups that didn’t have a timeout issue

### PR DESCRIPTION
**Pull request by noahl**
Closes #552 .

### CODE REVIEW

File: rho/ansible_utils.py [hunk 1]
```diff
@@ -69,16 +69,20 @@ def log_yaml_inventory(label, inventory):
...
-    vars_dict = alpha.get('vars')
+
     redact_key_list = ['ansible_become_pass', 'ansible_ssh_pass']
 
-    # pylint: disable=unused-variable
-    for host, host_dict in iteritems(hosts_dict):
-        host_dict = redact_dict(redact_key_list, host_dict)
+    groups = list(inventory.keys())
+
+    for group in groups:
```
Comment (elyezer) [inline rho/ansible_utils.py]: I see no reason to convert the inventory.keys into dict, am I missing something?

I think the for could be written as `for group in inventory.keys()` or even `for group in inventory`

File: rho/ansible_utils.py [hunk 2]
```diff
@@ -69,16 +69,20 @@ def log_yaml_inventory(label, inventory):
...
+        hosts_dict = inventory[group].get('hosts')
+        vars_dict = inventory[group].get('vars')
+
+        # pylint: disable=unused-variable
+        for host, host_dict in iteritems(hosts_dict):
```
Comment (elyezer) [inline rho/ansible_utils.py]: Rewriting this as `for host_dict in hosts_dict.values()` should avoid the pylint disable and won't need the host variable.

**Review by elyezer — COMMENTED** [review]
Leaving some comments, feel free to handle if they make sense.

Other than that it looks good. (I can't review 100% of the logic though, since I don't know much the code base)

**Review by chambridge — APPROVED** [review]
Will do some follow up PRs to try and more dynamically decide on the timeout and additionally tackle the heap memory issue. But this is a critical first step.

**Pull request merged**